sys/mips/gt64120.c

```diff
@@ -255,27 +253,12 @@ static intr_filter_t gt_pci_intr(void *data) {
   return IF_FILTERED;
 }
 
-#define MALTA_CORECTRL_SIZE (MALTA_CORECTRL_END - MALTA_CORECTRL_BASE + 1)
-#define MALTA_PCI0_MEMORY_SIZE                                                 \
-  (MALTA_PCI0_MEMORY_END - MALTA_PCI0_MEMORY_BASE + 1)
-
-DEVCLASS_DECLARE(pci);
-
 static int gt_pci_attach(device_t *pcib) {
   gt_pci_state_t *gtpci = pcib->state;
 
-  /* PCI I/O memory */
-  gtpci->pci_mem =
-    bus_alloc_resource(pcib, RT_MEMORY, 0, MALTA_PCI0_MEMORY_BASE,
-                       MALTA_PCI0_MEMORY_END, MALTA_PCI0_MEMORY_SIZE, 0);
-  /* PCI I/O ports 0x0000-0xffff, mapped into KVA */
-  gtpci->pci_io =
-    bus_alloc_resource(pcib, RT_MEMORY, 0, MALTA_PCI0_IO_BASE,
-                       MALTA_PCI0_IO_BASE + 0xffff, 0x10000, RF_ACTIVE);
-  /* GT64120 registers, mapped into KVA */
-  gtpci->corectrl =
-    bus_alloc_resource(pcib, RT_MEMORY, 0, MALTA_CORECTRL_BASE,
-                       MALTA_CORECTRL_END, MALTA_CORECTRL_SIZE, RF_ACTIVE);
+  gtpci->pci_mem = bus_alloc_resource_any(pcib, RT_MEMORY, 0, RF_NONE);
+  gtpci->pci_io = bus_alloc_resource_any(pcib, RT_MEMORY, 1, RF_ACTIVE);
+  gtpci->corectrl = bus_alloc_resource_any(pcib, RT_MEMORY, 2, RF_ACTIVE);
```

Comment:
Maybe it would be a good idea to make `define`s out of RIDs for each device? Rn this looks a tad too magical for my taste.

Reply:
Rids are mostly bar numbers. Every device must be aware of the number of its bars. For ISA devices,  they usually make only one allocation for io ports (rid = 0), and one for irq (rid = 0). This doesn't seem to be a problem to me (at least for now).